src/kofta_metadata.erl

```diff
@@ -254,16 +276,36 @@ make_request(State) ->
 
     TopicNames = dict:fetch_keys(ClientDict),
 
-    NewState = State#st{clients=dict:new(), last_batch=now()},
+    NewState = State#st{
+        clients=dict:new(),
+        broker_clients=dict:new(),
+        last_batch=now()
+    },
 
     case do_request(TopicNames, Host, Port) of
-        {ok, Topics} ->
-            lists:map(fun(Topic) ->
-                Clients = dict:fetch(Topic#topic.name, ClientDict),
-                lists:map(fun(Client) ->
-                    gen_fsm:reply(Client, {ok, Topic})
-                end, Clients)
-            end, Topics),
+        {ok, Brokers, Topics} ->
+            case dict:size(ClientDict) of
+                0 ->
```

Comment:
If you make a request with no topics, `Topics` will be all existing topics. If you make a request with more than 0 topics, `Topics` will only contain the topics you requested. So, if you didn't have that case clause, you'd be `foreach`ing over all topics and then the `dict:fetch` would throw exceptions.